test_runner/regress/test_timeline_archive.py

```diff
@@ -554,8 +554,33 @@ def worker():
                                 log.info(f"Timeline {state.timeline_id} is still active")
                                 shutdown.wait(0.5)
                             elif state.timeline_id in offloaded_ids:
-                                log.info(f"Timeline {state.timeline_id} is now offloaded")
-                                state.offloaded = True
+                                log.info(f"Timeline {state.timeline_id} is now offloaded in memory")
+
+                                # Hack: when we see something offloaded in the API, it doesn't guarantee that the offload
+                                # is persistent (it is marked offloaded first, then that is persisted to the tenant manifest).
+                                # So we wait until we see the manifest update before considering it offloaded, that way
+                                # subsequent checks that it doesn't revert to active on a restart will pass reliably.
+                                time.sleep(0.1)
```

Comment:
why sleep? if we depend on it, it's racy, if we don't depend on it, we can remove it.

I also don't see that we are waiting on the manifest update here.

Reply:
>why sleep? if we depend on it, it's racy, if we don't depend on it, we can remove it.

It's a small optimisation to reduce the chance that we go round a full iteration again (a longer sleep), and thereby increase the number of timeline lifecycles we explore in the test on average.  There's already a randomized sleep between iterations of the worker loop to de-correlate the timing of workers, so the sleep here shouldn't hurt determinism.